Modules/_asynciomodule.c

```diff
@@ -1601,11 +1601,24 @@ static void
 FutureIter_dealloc(futureiterobject *it)
 {
     PyTypeObject *tp = Py_TYPE(it);
-    asyncio_state *state = get_asyncio_state_by_def((PyObject *)it);
+
+    assert(_PyType_HasFeature((PyTypeObject *)tp, Py_TPFLAGS_HEAPTYPE));
+
+    PyHeapTypeObject *ht = (PyHeapTypeObject*)tp;
+    PyObject *module = ht->ht_module;
```

Comment:
Minor suggestion, we can make this a one-liner. Your call for which is more readable:

```suggestion
    PyObject *module = ((PyHeapTypeObject*)tp)->ht_module;
```